lib/domain/models/memo_execution.dart

```diff
@@ -9,30 +8,32 @@ class MemoExecution extends Equatable {
   MemoExecution({
     required this.started,
     required this.finished,
-    required this.question,
-    required this.answer,
-    required this.answeredDifficulty,
+    required this.rawQuestion,
+    required this.rawAnswer,
+    required this.markedDifficulty,
   })   : assert(started.isBefore(finished)),
-        assert(question.isNotEmpty),
-        assert(answer.isNotEmpty);
+        assert(rawQuestion.isNotEmpty),
+        assert(rawQuestion.first.isNotEmpty),
+        assert(rawAnswer.isNotEmpty),
+        assert(rawAnswer.first.isNotEmpty);
 
   final DateTime started;
   final DateTime finished;
   int get timeSpentInMillis => started.difference(finished).inMilliseconds;
 
-  final List<MemoBlock> question;
-  final List<MemoBlock> answer;
+  final List<Map<String, dynamic>> rawQuestion;
+  final List<Map<String, dynamic>> rawAnswer;
 
-  final MemoDifficulty answeredDifficulty;
+  final MemoDifficulty markedDifficulty;
 
   @override
-  List<Object?> get props => [started, finished, question, answer, answeredDifficulty];
+  List<Object?> get props => [started, finished, rawQuestion, rawAnswer, markedDifficulty];
 }
 
-/// Associates a `Collection.id` and all its executions of a particular memo, through its [memoId]
+/// Associates a `Collection.id` with all its executions of a particular `Memo`, through its [memoId]
 @immutable
-class MemoExecutions extends Equatable {
-  MemoExecutions({required this.memoId, required this.collectionId, required this.executions})
+class UniqueMemoExecutions extends Equatable {
```

Comment:
Why `Unique`? I mean, each execution is unique by definition, right?

Reply:
No, unique means that each memo can be executed any amount of times, although each represent a unique execution, meaning a reference to a single memo, while being executed one or multiple times.

Again, this is really confusing and I agree that this definition is completely subjective, and that is one of the reasons it was removed in a future PR.